cmd/onboard.go

```diff
@@ -187,65 +190,68 @@ TO1:
 			// A 25% plus or minus jitter is allowed by spec
 			select {
 			case <-ctx.Done():
-				return nil
+				return nil, ctx.Err()
 			case <-time.After(directive.Delay):
 			}
 		}
 	}
-	if to1d != nil {
-		for _, to2Addr := range to1d.Payload.Val.RV {
-			if to2Addr.DNSAddress == nil && to2Addr.IPAddress == nil {
-				slog.Error("Error: Both IP and DNS can't be null")
-				continue
-			}
+	if to1d == nil {
+		return nil, errors.Join(to1dErrs...)
+	}
+	for _, to2Addr := range to1d.Payload.Val.RV {
+		if to2Addr.DNSAddress == nil && to2Addr.IPAddress == nil {
+			slog.Error("Error: Both IP and DNS can't be null")
+			continue
+		}
 
-			var scheme, port string
-			switch to2Addr.TransportProtocol {
-			case protocol.HTTPTransport:
-				scheme, port = "http://", "80"
-			case protocol.HTTPSTransport:
-				scheme, port = "https://", "443"
-			default:
-				continue
-			}
-			if to2Addr.Port != 0 {
-				port = strconv.Itoa(int(to2Addr.Port))
-			}
+		var scheme, port string
+		switch to2Addr.TransportProtocol {
+		case protocol.HTTPTransport:
+			scheme, port = "http://", "80"
+		case protocol.HTTPSTransport:
+			scheme, port = "https://", "443"
+		default:
+			continue
```

Comment:
If not http or https then can we please log error , does this need any default handling ?